kazoo/tests/test_queue.py

```diff
@@ -66,88 +68,99 @@ def setUp(self):
             if ver[1] < 4:
                 skip = True
         if skip:
-            raise SkipTest("Must use Zookeeper 3.4 or above")
+            pytest.skip("Must use Zookeeper 3.4 or above")
 
     def _makeOne(self):
         path = "/" + uuid.uuid4().hex
         return self.client.LockingQueue(path)
 
     def test_queue_validation(self):
```

Comment:
this could also be changed to a pytest parametrized test

Reply:
Not sure what you mean by that. Can you show me an example? A link to some tests leveraging that?

Reply:
https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions

```
@pytest.mark.parametrize("input,expected", [
    ({}, TypeError),
    (b"one", b"100", TypeError),
    (etc, etc),
])
def test_queue_validation(self, input, expected):
queue = self._makeOne()
with pytest.raises(expected):
    queue.put(input)
```

Reply:
I really struggled to get this to work with classes.
After much digging, it looks like this is an example of how it would work:
https://doc.pytest.org/en/3.0.5/example/parametrize.html#a-quick-port-of-testscenarios

Now, all that metaclass overhead really muddies the water in my opinion. Do you really want me to go down that route?
Alternatively, did I miss something obvious trying to get `pytest.mark.parametrize()` to work with `unittest.TestCase`?

Reply:
Ah, I forgot that these use `unittest.TestCase`... yeah, probably best to skip it for now. Perhaps at some point in the future we can drop `unittest` completely in favor `pytest` fixtures, as I find them a lot more powerful but that's beyond the scope of this PR... rather get this across the finish line.